index.bs

```diff
@@ -688,36 +688,23 @@ The Working Group has started documenting ethical issues associated with using M
 # Programming Model # {#programming-model}
 ## Overview ## {#programming-model-overview}
 
-At the heart of neural networks is a computational graph of mathematical operations.
+At the heart of neural networks is a <dfn>computational graph</dfn> of mathematical operations.
 These operations are the building blocks of modern machine learning technologies in
 computer vision, natural language processing, and robotics.
 The WebNN API is a specification for constructing, compiling, and executing computational
 graphs of neural networks.
 
 The {{MLGraph}} interface represents a compiled computational graph that is immutable (that is, a model).
 
-The {{MLGraphBuilder}} interface serves as a builder (factory) to create an {{MLGraph}}.
-An {{MLOperand}} is a representation of data that flows within the computational graph,
-which include input-values for inference, constants (including trained weights)
-used for inference, intermediate values (often referred to as activations) computed
-during inference, as well as the output values of inference.
-At inference time, every {{MLOperand}} will be bound to a tensor (the actual data).
+The {{MLGraphBuilder}} interface serves as a builder (factory) to construct a [=computational graph=] (its <dfn for=MLGraphBuilder>graph</dfn>) that is then compiled to create an {{MLGraph}}.
 
-The {{MLGraphBuilder}} interface enables the creation of {{MLOperand}}s.
-A key part of the {{MLGraphBuilder}} interface are the operations (such as
-{{MLGraphBuilder}}.{{MLGraphBuilder/gemm()}} and {{MLGraphBuilder}}.{{MLGraphBuilder/softmax()}}). The operations have a functional
-semantics, with no side effects.
-Each operation invocation conceptually returns a distinct new value, without
-changing the value of any other {{MLOperand}}. 
+In WebNN, a [=computational graph=] is composed of <dfn>operators</dfn> which act on data, and are the nodes of the graph. {{MLOperand}}s are a representation of data that flows within the computational graph, and are the edges of the graph. {MLOperand}}s include a [=computational graph=]'s <dfn for="computational graph">input</dfn> values for inference, <dfn for="computational graph">constants</dfn> (including trained weights) used for inference, intermediate values (often referred to as activations) computed during inference, as well as the output values of inference. An [=operator=]'s <dfn for=operator>input</dfn> is one or more {{MLOperand}}s. An [=operator=]'s <dfn for=operator>output</dfn> is one or more {{MLOperand}}s. [=Operators=] have operator-specific parameters that control their behavior, which can include zero or more <dfn for=operator lt="activation|activation function">activation functions</dfn>, which are {{MLActivation}}s.
```

Comment:
The only remaining thing is whether we should define <dfn for="computational graph">outputs</dfn>. Because it is used by `build()` method, we probably can defer it to #457.

Reply:
Agreed - I didn't want to define an unused term just yet, but that was the intention.